server/src/main/java/com/linecorp/centraldogma/server/internal/api/ProjectServiceV1.java

```diff
@@ -93,134 +94,62 @@ public ProjectServiceV1(ProjectManager projectManager, CommandExecutor executor,
+     * DELETE /projects/{projectName}
      *
-     * <p>Remove a user from the member of the project.
+     * <p>Removes a project.
      */
-    @Delete("/projects/{projectName}/members/{user}")
-    public CompletionStage<ProjectInfo> removeMember(@Param("projectName") String projectName,
-                                                     @Param("user") String user,
-                                                     @RequestObject Author author) {
-        return mds.removeMember(projectName, author, new User(user));
+    @Delete("/projects/{projectName}")
+    @Decorator(ProjectOwnersOnly.class)
+    public CompletionStage<Void> removeProject(@RequestObject Project project,
+                                               @RequestObject Author author) {
+        return mds.removeProject(author, project.name())
+                  .thenCompose(unused -> execute(Command.removeProject(author, project.name())))
+                  .handle(HttpApiUtil::throwUnsafelyIfNonNull);
     }
 
     /**
-     * PATCH /projects/{projectName}/members/{user}?role={role}
+     * PATCH /projects/{projectName}
      *
-     * <p>Changes the role of a user.
+     * <p>Patches a project with the JSON_PATCH. Currently, only unremove project operation is supported.
      */
     @ConsumeType("application/json-patch+json")
-    @Patch("/projects/{projectName}/members/{user}")
-    public CompletionStage<ProjectInfo> changeMemberRole(@Param("projectName") String projectName,
-                                                         @Param("user") String user,
-                                                         @Param("role") String role,
-                                                         @RequestObject Author author) {
-        return mds.changeMemberRole(projectName, author, new User(user),
-                                    ProjectRole.of(role));
-    }
-
-    /**
-     * POST /projects/{projectName}/tokens/{appId}
-     *
-     * <p>Adds a token to the project.
-     */
-    @Post("/projects/{projectName}/tokens/{appId}")
-    public CompletionStage<ProjectInfo> addToken(@Param("projectName") String projectName,
-                                                 @Param("appId") String appId,
-                                                 @Param("role") Optional<String> role,
-                                                 @RequestObject Author author) {
-        return mds.addToken(projectName, author, appId,
-                            role.map(ProjectRole::of).orElse(ProjectRole.MEMBER));
-    }
-
-    /**
-     * DELETE /projects/{projectName}/tokens/{appId}
-     *
-     * <p>Remove a token from the project.
-     */
-    @Delete("/projects/{projectName}/tokens/{appId}")
-    public CompletionStage<ProjectInfo> removeToken(@Param("projectName") String projectName,
-                                                    @Param("appId") String appId,
+    @Patch("/projects/{projectName}")
+    @Decorator(AdministratorsOnly.class)
```

Comment:
If the owner of a project removes the project by mistake, does he/she have to ask the administrator to bring it back?

Reply:
That's a good question. I wonder if the project owner is allowed to unremove his or her project.

Reply:
As we discussed in LINE chat room, it'd be better to leave this as it is. A metadata is stored in a `meta` repository of each project so the metadata is not able to be accessed if a project is removed, by default. That means we have no chance to find a role of a user in our authentication decorator, for now. If we want to read a metadata from a removed project, we need to make some code more, but it's not an important specification now.